src/osd/OSDMap.cc

```diff
@@ -3081,11 +3014,12 @@ int OSDMap::build_simple_crush_map_from_conf(CephContext *cct,
   // add osds
   vector<string> sections;
   conf->get_all_sections(sections);
-  for (vector<string>::iterator i = sections.begin(); i != sections.end(); ++i) {
-    if (i->find("osd.") != 0)
+
+  for (auto &elem : sections) {
```

Comment:
@tchaikov Why there are two sections vectors here ? It is confusing. Can I change the second one to 'newsections' ? Will it break the logic ?